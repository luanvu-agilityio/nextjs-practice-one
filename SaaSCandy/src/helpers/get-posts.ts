import { blogService } from '@/api';
import { BlogPost } from '@/types/blog-post';

export const getPostBySlug = async (slug: string): Promise<BlogPost | null> => {
  return await blogService.getPostBySlug(slug);
};

export const getAllPosts = async (): Promise<BlogPost[]> => {
  return await blogService.getAllPosts();
};

export const getRecentPosts = async (
  limit: number = 3
): Promise<BlogPost[]> => {
  const posts = await blogService.getAllPosts();
  return posts
    .sort((a, b) => new Date(b.date).getTime() - new Date(a.date).getTime())
    .slice(0, limit);
};

export const getPostsByTag = async (tag: string): Promise<BlogPost[]> => {
  return await blogService.getPostsByTag(tag);
};
