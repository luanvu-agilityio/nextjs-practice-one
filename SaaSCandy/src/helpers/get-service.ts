import { services } from '@/constants';
import { Service } from '@/types';

export function getAllServices(): Service[] {
  return services;
}

export function getServiceBySlug(slug: string): Service | null {
  return services.find(service => service.slug === slug) || null;
}

export function getServiceSlugs(): string[] {
  return services.map(service => service.slug);
}
