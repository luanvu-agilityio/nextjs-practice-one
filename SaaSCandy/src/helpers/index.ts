export * from './get-service';
export * from './get-posts';
