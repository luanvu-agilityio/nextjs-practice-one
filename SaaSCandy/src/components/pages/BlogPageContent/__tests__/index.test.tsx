import { render } from '@testing-library/react';
import { BlogPageContent } from '..';
import { getAllPosts } from '@/helpers';

jest.mock('@/helpers');
jest.mock('@/components/BlogList', () => ({
  BlogList: () => <div>Mocked BlogList</div>,
}));

describe('BlogPage Component', () => {
  const mockPosts = [
    {
      slug: 'post-1',
      title: 'Test Post 1',
      excerpt: 'Excerpt 1',
      date: '2025-01-01',
      image: '/test1.jpg',
      tags: ['test'],
      comments: 0,
    },
    {
      slug: 'post-2',
      title: 'Test Post 2',
      excerpt: 'Excerpt 2',
      date: '2025-01-02',
      image: '/test2.jpg',
      tags: ['test'],
      comments: 5,
    },
  ];

  beforeEach(() => {
    (getAllPosts as jest.Mock).mockResolvedValue(mockPosts);
  });

  it('matches snapshot', () => {
    const { container } = render(<BlogPageContent />);
    expect(container).toMatchSnapshot();
  });
});
