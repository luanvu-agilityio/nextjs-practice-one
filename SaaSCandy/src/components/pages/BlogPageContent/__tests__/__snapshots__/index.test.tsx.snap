// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BlogPage Component matches snapshot 1`] = `
<div>
  <section
    class="py-12 sm:py-16 bg-white"
  >
    <div
      class="max-w-[1296px] mx-auto px-4 sm:px-6 lg:px-0 "
    >
      <div
        class="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-4 sm:gap-6"
      >
        <div>
          Mocked BlogList
        </div>
      </div>
    </div>
  </section>
</div>
`;
